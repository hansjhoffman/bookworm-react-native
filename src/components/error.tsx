import React from "react";
import { Text } from "react-native";

type Props = {
  message?: string;
} & typeof defaultProps;

const defaultProps = {
  message: "Ah ah ah, you didn't say the magic word.",
};

export function Error({ message }: Props) {
  return <Text>{message}</Text>;
}

Error.defaultProps = defaultProps;
