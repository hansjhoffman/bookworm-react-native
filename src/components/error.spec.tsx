import React from "react";
import { render } from "@testing-library/react-native";

import { Error } from "./error";

describe("[component] Error", function() {
  it("should handle custom error message", function() {
    let errorMsg: string = "This is a test message!";

    const { getByText } = render(<Error message={errorMsg} />);

    getByText(errorMsg);
  });

  it("should match snapshot", function() {
    const { baseElement } = render(<Error />);

    expect(baseElement).toMatchSnapshot();
  });
});
