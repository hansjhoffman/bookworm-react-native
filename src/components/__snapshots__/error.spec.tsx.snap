// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component] Error should match snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text>
      Ah ah ah, you didn't say the magic word.
    </Text>
  </View>
</View>
`;
